emma-language/src/main/scala/org/emmalanguage/api/ScalaSeq.scala

```diff
@@ -89,6 +90,23 @@ class ScalaSeq[A] private[api](private[api] val rep: Seq[A]) extends DataBag[A]
     sample.toVector
   }
 
+  def split(fractions: Double*)(seed: Long = 631431513L): Array[DataBag[A]] = {
+    val n = fractions.length
+    // normalize fractions
+    val normalized = fractions.map(_ / fractions.sum)
+    // compute cdf
+    val cdf = normalized.scanLeft(0.0)(_ + _)
+
+    // generate random assignment based on CDF
+    val random = util.RanHash(seed)
```

Comment:
You need to seed the `RanHash` based on the current element index. See how this is done in `sample`.